reconcile/utils/terrascript_aws_client.py

```diff
@@ -781,7 +781,9 @@ def populate_route53(self, desired_state, default_ttl=300):
             acct_name = zone["account_name"]
 
             # Ensure zone is in the state for the given account
-            zone_id = safe_resource_id(f"{zone['name']}")
+            # zone_id refers to the terraform identifier, not the AWS
+            # route 53 zone ID that is generated after the zone is created
+            zone_id = safe_resource_id(f"{zone['id']}")
             zone_values = {
                 "name": zone["name"],
                 "vpc": zone.get("vpc"),
```

Comment:
And then on line `792` (I'm having trouble selecting it, replace `zone_id` with `zone_res_name`.